### PR TITLE
Add Linux setup for flashing

### LINKED ISSUE #1959: Flashing fails on Linux
**Describe the bug**
Using Arch Linux with a user added to the `dialout` group, I cannot flash a game to my Sprig.

**To Reproduce**
1. Plug in your Sprig.
2. Make sure your user is part of the `dialout` group (`usermod username -aG dialout`)
3. In a Chromium based browser (I used Chromium `126.0.6478.126 (Official Build) (64-bit) `) attempt to flash the game after selecting the Pico.
4. Nothing happens on the Sprig. The console shows that the serial port could not be opened.

**Expected behavior**
Game to flash successfully like on macOS

**Actual behavior**  
Warning triangle displayed next to the button and the logs shown below in the console.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by DillonB07**
This expands the document about uploading games to Sprig to include more details about what may be required on Linux systems.
I've added notes about user groups and flatpaks.

This PR is created from my issues with flashing on Linux and discussion in [this Slack thread](https://hackclub.slack.com/archives/C02UN35M7LG/p1720893590000369). Also closes #1959

Also, I used the new GitHub note syntax. I wasn't sure whether to go back and change the other notes or keep it consistent with the old syntax so maybe that should be updated.

### CODE REVIEW

File: docs/UPLOAD.md [hunk 1]
```diff
@@ -49,6 +49,42 @@ cp -X ~/Downloads/pico-os.uf2 /Volumes/RPI-RP2
...
+sudo usermod -aG dialout orpheus
+```
+> [!NOTE]
+> You will likely be prompted for your password. Don't worry if you can't see it - that's normal! Just type it out as normal and press enter
+
+**Make sure to reboot your system after making this change** and you should be good to get flashing!
+
+### Flatpaks - I play with sandboxes!
+
+Is your browser installed as a Flatpak? This means that you need to do more setup becauase it's running in a sandbox so it likely won't have access to serial ports. To fix this, run the following override command replacing the flatpak name as appropriate:
```
Comment (OtterCodes101) [inline docs/UPLOAD.md:80]: because isn't spelled right

**Review by recursiveforte — APPROVED** [review]
LGTM!

File: docs/UPLOAD.md [hunk 2]
```diff
@@ -44,6 +44,42 @@ cp -X ~/Downloads/pico-os.uf2 /Volumes/RPI-RP2
...
+> If you're using Linux, check the Linux section below because you probably need more setup!
+
+
+## But I use Linux!
+
+Chances are, you need to do some additional setup to flash your game on Linux. But don't fear - it's straightforward!
+
+### Join the group!
+
+For most Linux distributions, you need to add your user to the `dialout` group. If this group exists, or it doesn't work, add your user to the `uucp` group instead.
```
Comment (lafkpages) [inline docs/UPLOAD.md:60]: Are you sure you didn't mean "If this group doesn't exist"? Lol this PR showed up in my GitHub feed this morning and that just didn't really sound right

Comment (DillonB07) [reply docs/UPLOAD.md:60]: Nice catch!

Comment (DillonB07) [reply docs/UPLOAD.md:60]: Resolved with #2200

**Pull request merged**